compiler/src/iree/compiler/DispatchCreation/CollapseDimensions.cpp

```diff
@@ -134,23 +133,36 @@ static SmallVector<ReassociationIndices> getCollapsibleLoops(Operation *op) {
   //    found in all maps. If so, add to sequence (to get a sequence of 3)
   //    and repeat till the last element of sequence and the next result
   //    expression is not found as a sequence in all maps.
-  for (auto nextExpr :
-       fusionInterfaceOp.getIndexingMapsArray().front().getResults()) {
-    unsigned position = cast<AffineDimExpr>(nextExpr).getPosition();
-    if (!range.empty()) {
-      if (!hasAllMapsSameSequence(preExpr, nextExpr) ||
-          !hasSameIteratorType(preExpr, nextExpr)) {
+
+  llvm::DenseSet<unsigned> handledExprs;
+  for (auto map : fusionInterfaceOp.getIndexingMapsArray()) {
+    ReassociationIndices range;
+    AffineExpr preExpr;
+    for (auto nextExpr : map.getResults()) {
+      unsigned position = cast<AffineDimExpr>(nextExpr).getPosition();
+      if (handledExprs.contains(position)) {
+        preExpr = nullptr;
         if (range.size() > 1) {
           contiguousLoops.push_back({range.begin(), range.end()});
         }
         range.clear();
+        continue;
+      } else if (!range.empty()) {
+        if (!hasAllMapsSameSequence(preExpr, nextExpr) ||
+            !hasSameIteratorType(preExpr, nextExpr)) {
+          if (range.size() > 1) {
+            contiguousLoops.push_back({range.begin(), range.end()});
```

Comment:
why not just `push_back(range)`? (I have the same question for line 146.)

Reply:
Good catch, it should be `push_back(range)`